src/setup.rs

```diff
@@ -93,6 +93,31 @@ pub(crate) fn setup() {
         rcc.ahb1rstr.modify(|_, w| w.ethmacrst().set_bit());
         rcc.ahb1rstr.modify(|_, w| w.ethmacrst().clear_bit());
     });
+
+    #[cfg(feature = "stm32f1xx-hal")]
+    cortex_m::interrupt::free(|_| unsafe {
+        let afio = &*crate::stm32::AFIO::ptr();
+        let rcc = &*crate::stm32::RCC::ptr();
+
+        // select MII or RMII mode
+        // 0 = MII, 1 = RMII
+        afio.mapr.modify(|_, w| w.mii_rmii_sel().set_bit());
```

Comment:
Shouldn't you enable the `AFIO` clock before doing this ? And shouldn't you check to see if the MAC is enabled like the other implementations ?

Another thing, I would prefer this to use bitbanding since it's available on the F1.